lib/mongo/socket/ssl.rb

```diff
@@ -137,7 +137,7 @@ def initialize(host, port, host_name, timeout, family, options = {})
       #
       # @since 2.0.0
       def connect!
-        Timeout.timeout(options[:connect_timeout], Error::SocketTimeoutError) do
+        Timeout.timeout(options[:connect_timeout], Error::SocketTimeoutError, 'the socket took too long to connect') do
```

Comment:
Can you please add the timeout into the message, e.g. `The socket took over #{options[:connect_timeout]} seconds to connect` (and capitalize the first letter).